src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py

```diff
@@ -2,98 +2,125 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
-
-from azure.cli.core.commands.transform import _parse_id
+# pylint: disable=line-too-long, protected-access
 from msrestazure.tools import parse_resource_id
 from knack.log import get_logger
 from knack.util import CLIError
-
-from .._client_factory import cf_monitor
-from ..util import gen_guid
+from azure.cli.core.commands.transform import _parse_id
+from azure.cli.command_modules.network.custom import _convert_to_snake_case
+from azure.cli.command_modules.monitor.util import gen_guid
 
 logger = get_logger(__name__)
 CLONED_NAME = "cloned-{}-{}"
 
 
-def _gen_metrics_alert_rules_clone_list(monitor_client, source_resource, target_resource):
-    alert_rules = list(monitor_client.metric_alerts.list_by_subscription())
+def _get_metrics_alert_rules_clone_list(cmd, source_resource, target_resource):
+    subscription_id = parse_resource_id(source_resource)['subscription']
+    from ..aaz.latest.monitor.metrics.alert import List
+    alert_rules = List(cli_ctx=cmd.cli_ctx)(command_args={"subscription_id": subscription_id})
```

Comment:
Why the previous implementation does not use subsciption_id?
As I remember, the subscription is fetch from global by default, did you try the subscription_id can be passed in as an valid value or not?

Reply:
Previously this cmd uses different monitor client with different subscripionid
Passing customerized subscriptionid as a parameter is supported from AAZCommandCtx